RIOT-OS/RIOT
# dist/tools/cppcheck: fix all Cppcheck 1.82 errors

### Contribution description

`cppcheck` errors are making merging #12558 more difficult so this PR fixes this.

### Testing procedure

- Run locally: `./dist/tools/cppcheck/check.sh` no errors should appear
- Green Murdock

Please take a closer look at 1e671ac and c2f593f,

1e671ac I'm not sure if `int sig = -1` is the best initial value for `sig`, although shouldn't really matter since its updated in the while loop..

c2f593f is because `RTT_FREQUENCY` doesn't have a default value so it results in a division by 0. Maybe there is a best way than duplicating the warning.

If preferred I can squash all commits fixing same errors.

### Issues/PRs references

#12558 can benefit from this.
